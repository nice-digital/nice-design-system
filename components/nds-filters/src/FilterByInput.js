import React, { Component } from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import { Button } from "@nice-digital/nds-button";
import { Input } from "@nice-digital/nds-input";

import "./../scss/filter-input.scss";

export class FilterByInput extends Component {
	constructor(props) {
		super(props);

		const { collapseByDefault = false } = this.props;

		this.state = {
			isExpanded: !collapseByDefault,
			canUseDOM: false
		};

		this.handleTitleClick = this.handleTitleClick.bind(this);
	}

	componentDidMount() {
		this.setState({ canUseDOM: true });
	}

	handleTitleClick(e) {
		e.preventDefault();

		this.setState(prevState => ({
			isExpanded: !prevState.isExpanded
		}));
	}

	render() {
		const {
				className,
				type = "text",
				label,
				name,
				buttonLabel = "Filter",
				inputProps,
				...rest
			} = this.props,
			{ isExpanded } = this.state;

		return (
			<div className={classnames("inputFilterBox", className)} {...rest}>
				<h3 className="inputFilterBox__heading">
					{this.state.canUseDOM ? (
						<button
							type="button"
							aria-expanded={isExpanded}
							aria-controls={`inputFilter-${name}`}
							onClick={this.handleTitleClick}
						>
							{label}
						</button>
					) : (
						<>{label}</>
					)}
				</h3>
				<div
					id={`inputFilter-${name}`}
					aria-hidden={!isExpanded}
					className="inputFilterBox__controls"
				>
					<Input type={type} label={label} name={name} {...inputProps} />
					<Button type="submit" className="ml--0 mb--0">
						{buttonLabel}
					</Button>
				</div>
			</div>
		);
	}
}

FilterByInput.propTypes = {
	inputType: PropTypes.string,
	type: PropTypes.string,
	buttonLabel: PropTypes.string,
	label: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
	collapseByDefault: PropTypes.bool,
	className: PropTypes.string,
	inputProps: PropTypes.any
};

export default FilterByInput;
