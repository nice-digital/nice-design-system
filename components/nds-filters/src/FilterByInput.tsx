import React, { useState, useEffect } from "react";
import classnames from "classnames";
import { Button } from "@nice-digital/nds-button";
import { Input } from "@nice-digital/nds-input";
import ChevronDown from "@nice-digital/icons/lib/ChevronDown";

import "./../scss/filter-input.scss";

export interface FilterByInputProps {
	[prop: string]: unknown;
	label: string;
	name: string;
	buttonLabel?: string;
	className?: string;
	collapseByDefault?: boolean;
	type?:
		| "color"
		| "date"
		| "datetime-local"
		| "email"
		| "file"
		| "hidden"
		| "image"
		| "month"
		| "number"
		| "range"
		| "password"
		| "search"
		| "tel"
		| "text"
		| "time"
		| "url"
		| "week";
	inputProps?: object;
	headingLevel?: 3 | 4 | 5 | 6;
}

export const FilterByInput: React.FC<FilterByInputProps> = (
	props: FilterByInputProps
) => {
	const {
		className,
		type = "text",
		label,
		name,
		buttonLabel = "Filter",
		inputProps,
		headingLevel = 3,
		collapseByDefault = false,
		...rest
	} = props;

	const [isExpanded, setIsExpanded] = useState(!collapseByDefault);
	const [canUseDOM, setCanUseDOM] = useState(false);

	useEffect(() => {
		setCanUseDOM(true);
	}, []);

	const handleTitleClick = (e: React.MouseEvent<HTMLElement>) => {
		e.preventDefault();
		setIsExpanded(!isExpanded);
	};

	const HeadingLevel = `h${headingLevel}` as keyof JSX.IntrinsicElements;

	const filteredProps = Object.assign({}, ...[rest]);

	const propsToRemoveFromDom = ["collapseByDefault"];

	propsToRemoveFromDom.forEach((prop) => {
		delete filteredProps[prop];
	});

	return (
		<div
			className={classnames("inputFilterBox", className)}
			data-component="input-filter"
			{...filteredProps}
		>
			<HeadingLevel className="inputFilterBox__heading">
				{canUseDOM ? (
					<button
						type="button"
						aria-expanded={isExpanded}
						aria-controls={`inputFilter-${name}`}
						onClick={handleTitleClick}
					>
						<ChevronDown
							className={classnames([
								"filter-group__heading-icon",
								isExpanded && "filter-group__heading-icon--expanded"
							])}
						/>
						{label}
					</button>
				) : (
					<>{label}</>
				)}
			</HeadingLevel>
			<div
				id={`inputFilter-${name}`}
				aria-hidden={!isExpanded}
				className="inputFilterBox__controls"
			>
				<Input type={type} label={label} name={name} {...inputProps} />
				<Button type="submit" className="ml--0 mb--0">
					{buttonLabel}
				</Button>
			</div>
		</div>
	);
};

FilterByInput.displayName = "FilterByInput";
