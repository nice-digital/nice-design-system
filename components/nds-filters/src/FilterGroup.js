import React, { Component } from "react";
import PropTypes from "prop-types";

import "./../scss/filter-panel.scss";

export class FilterGroup extends Component {
	constructor(props) {
		super(props);

		this.state = {
			isExpanded: true,
			canUseDOM: false
		};

		this.handleTitleClick = this.handleTitleClick.bind(this);
	}

	componentDidMount() {
		this.setState({ canUseDOM: true });
	}

	handleTitleClick() {
		this.setState(prevState => ({
			isExpanded: !prevState.isExpanded
		}));
	}

	render() {
		return <div className="filter-group">TODO</div>;
	}
}

FilterGroup.propTypes = {};

FilterGroup.defaultProps = {};

export default FilterGroup;
