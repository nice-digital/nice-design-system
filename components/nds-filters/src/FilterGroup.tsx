import React, { useEffect, useState } from "react";
import classnames from "classnames";
import ChevronDown from "@nice-digital/icons/lib/ChevronDown";

import { slugify } from "@nice-digital/nds-core";

import "./../scss/filter-group.scss";

export interface FilterGroupProps {
	[prop: string]: unknown;
	heading: string;
	id?: string;
	selectedCount?: number;
	collapseByDefault?: boolean;
	children: React.ReactNode;
	className?: string;
	headingLevel?: 3 | 4 | 5 | 6;
}

export const FilterGroup: React.FC<FilterGroupProps> = (
	props: FilterGroupProps
) => {
	const {
		collapseByDefault = false,
		selectedCount,
		id,
		heading,
		headingLevel = 3,
		children,
		className,
		...rest
	} = props;
	const [isExpanded, setIsExpanded] = useState(
		(selectedCount && selectedCount > 0) || !collapseByDefault
	);
	const [canUseDOM, setCanUseDOM] = useState(false);

	useEffect(() => {
		setCanUseDOM(true);
	}, []);

	const handleTitleClick = (e: React.MouseEvent<HTMLElement>) => {
		e.preventDefault();
		setIsExpanded(!isExpanded);
	};

	const groupId = id || slugify(heading);

	let numSelected = null;
	if (selectedCount && selectedCount > 0) {
		numSelected = (
			<span className="filter-group__count">
				<span className="visually-hidden">(</span>
				{selectedCount} selected<span className="visually-hidden">)</span>
			</span>
		);
	}

	const HeadingLevel = `h${headingLevel}` as keyof JSX.IntrinsicElements;

	const groupHeadingElement = (
		<>
			<span id={`group-heading-${groupId}`}>{heading} </span>
			{numSelected}
		</>
	);

	const clonedChildren = React.Children.map(children, (child: any) => {
		if (React.isValidElement(child as React.ReactElement)) {
			return React.cloneElement(child, {
				groupId,
				groupHeading: heading
			});
		}
	});

	const filteredProps = Object.assign({}, ...[rest]);

	const propsToRemoveFromDom = ["collapseByDefault"];

	propsToRemoveFromDom.forEach((prop) => {
		delete filteredProps[prop];
	});

	return (
		<div className={classnames("filter-group", className)} {...filteredProps}>
			<HeadingLevel className="filter-group__heading">
				{canUseDOM ? (
					<button
						type="button"
						aria-expanded={isExpanded}
						aria-controls={`group-${groupId}`}
						onClick={handleTitleClick}
					>
						<ChevronDown
							className={classnames([
								"filter-group__heading-icon",
								isExpanded && "filter-group__heading-icon--expanded"
							])}
						/>
						{groupHeadingElement}
					</button>
				) : (
					<>{groupHeadingElement}</>
				)}
			</HeadingLevel>
			<fieldset
				id={`group-${groupId}`}
				aria-hidden={!isExpanded}
				className="filter-group__options"
			>
				<legend>{heading}</legend>
				{clonedChildren}
			</fieldset>
		</div>
	);
};

FilterGroup.displayName = "FilterGroup";
