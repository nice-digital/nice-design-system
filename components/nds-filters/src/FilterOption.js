import React, { Component } from "react";
import PropTypes from "prop-types";

import { slugify } from "@nice-digital/nds-core";

import "./../scss/filter-option.scss";

export class FilterOption extends Component {
	constructor(props) {
		super(props);

		this.state = {
			selected: this.props.isSelected
		};

		this.handleCheckboxChange = this.handleCheckboxChange.bind(this);
	}

	UNSAFE_componentWillReceiveProps(nextProps) {
		this.setState({
			selected: nextProps.isSelected
		});
	}

	handleCheckboxChange() {
		this.setState(
			prevState => ({
				selected: !prevState.selected
			}),
			() => {
				this.props.onChanged && this.props.onChanged();
			}
		);
	}

	render() {
		const {
			groupId,
			groupHeading,
			value,
			children,
			/* eslint-disable */
			// declaring unused props to avoid spreading across the dom
			onChanged,
			isSelected,
			/* eslint-enable */
			...rest
		} = this.props;

		const { selected } = this.state;

		const slugifiedValue = value ? slugify(value) : slugify(children);

		return (
			<label
				htmlFor={`filter_${groupId}_${slugifiedValue}`}
				className="filter-option"
				{...rest}
			>
				<input
					id={`filter_${groupId}_${slugifiedValue}`}
					type="checkbox"
					name={groupId}
					value={value || children}
					checked={selected}
					title={`${groupHeading} - ${children}`}
					onChange={this.handleCheckboxChange}
				/>
				<span className="filter-option__text">{children}</span>
			</label>
		);
	}
}

FilterOption.propTypes = {
	groupId: PropTypes.string,
	groupHeading: PropTypes.string,
	isSelected: PropTypes.bool.isRequired,
	children: PropTypes.string.isRequired,
	value: PropTypes.string,
	onChanged: PropTypes.func
};
