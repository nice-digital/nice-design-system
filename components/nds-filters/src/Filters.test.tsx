import React from "react";
import { render, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { FilterSummary, FilterSummaryProps } from "./FilterSummary";
import { FilterPanel, FilterPanelProps } from "./FilterPanel";
import { FilterGroup, FilterGroupProps } from "./FilterGroup";
import { FilterOption, FilterOptionProps } from "./FilterOption";
import { FilterByInput, FilterByInputProps } from "./FilterByInput";

const aFunction = jest.fn();

const filterSummaryProps: FilterSummaryProps = {
	sorting: [
		{ label: "Active", active: true },
		{ label: "Button", onSelected: aFunction }
	],
	activeFilters: [
		{
			label: "Another filter",
			onClick: aFunction
		}
	],
	children: "Showing results 1 to 10 of 1209"
};

const filterPanelProps: FilterPanelProps = {
	heading: "Some filters",
	fallback: {
		method: "GET",
		action: "/submit"
	},
	onSubmit: aFunction,
	children: "A filter panel"
};

const filterGroupProps: FilterGroupProps = {
	heading: "Some filters",
	children: "A filter group"
};

const filterOptionProps: FilterOptionProps = {
	isSelected: true,
	onChanged: aFunction,
	children: "A filter option"
};

const filterByInputProps: FilterByInputProps = {
	label: "Filter by name",
	name: "Filter by name"
};

describe("@nice-digital/nds-filters", () => {
	// describe("Crash test", () => {
	// 	it("should render filter summary without crashing", () => {
	// 		render(<FilterSummary {...filterSummaryProps} />);
	// 	});

	// 	it("should render filter panel without crashing", () => {
	// 		render(<FilterPanel {...filterPanelProps} />);
	// 	});

	// 	it("should render filter group without crashing", () => {
	// 		render(<FilterGroup {...filterGroupProps} />);
	// 	});

	// 	it("should render a filter option without crashing", () => {
	// 		render(<FilterOption {...filterOptionProps} />);
	// 	});

	// 	it("should render a filter by input component without crashing", () => {
	// 		render(<FilterByInput {...filterByInputProps} />);
	// 	});

	// 	it("should match snapshot", () => {
	// 		const wrapper = render(
	// 			<>
	// 				<FilterSummary {...filterSummaryProps} />
	// 				<FilterPanel {...filterPanelProps}>
	// 					<FilterGroup {...filterGroupProps}>
	// 						<FilterOption {...filterOptionProps}>A filter</FilterOption>
	// 					</FilterGroup>
	// 					<FilterByInput {...filterByInputProps} />
	// 				</FilterPanel>
	// 			</>
	// 		);
	// 		expect(wrapper).toMatchSnapshot();
	// 	});
	// });

	// describe("FilterSummary component", () => {
	// 	it("should preselect whichever sorting option has an active property", () => {
	// 		const callback = () => {
	// 			console.log("Callback!");
	// 		};

	// 		const sorting = [
	// 			{ label: "Item 1", value: "item-1", onSelected: callback },
	// 			{ label: "Item 2", value: "item-2", onSelected: callback },
	// 			{
	// 				label: "Item 3 (active)",
	// 				value: "item-3",
	// 				onSelected: callback,
	// 				active: true
	// 			}
	// 		];
	// 		const { container } = render(
	// 			<FilterSummary
	// 				sorting={sorting}
	// 				activeFilters={filterSummaryProps.activeFilters}
	// 			>
	// 				Showing results 1 to 10 of 1209
	// 			</FilterSummary>
	// 		);
	// 		const selectedItem = container.querySelector('[value="item-3"]');
	// 		expect(selectedItem?.hasAttribute("selected")).toBe(true);
	// 	});
	// });

	// describe("FilterByInput component", () => {
	// 	it("should pass inputProps down to the input ", () => {
	// 		const { container } = render(
	// 			<>
	// 				<FilterPanel {...filterPanelProps}>
	// 					<FilterByInput
	// 						{...filterByInputProps}
	// 						inputProps={{
	// 							hint: "enter your search term here"
	// 						}}
	// 					/>
	// 				</FilterPanel>
	// 			</>
	// 		);
	// 		const input = container.querySelector("p");
	// 		expect(input?.textContent).toBe("enter your search term here");
	// 	});
	// });

	// describe("FilterGroup component", () => {
	// 	it("should collapse the filter group when set to collapse by default", () => {
	// 		const { container } = render(
	// 			<FilterGroup {...filterGroupProps} collapseByDefault={true}>
	// 				A filter group
	// 			</FilterGroup>
	// 		);
	// 		const fieldset = container.querySelector("fieldset");
	// 		expect(fieldset?.getAttribute("aria-hidden")).toBe("true");
	// 	});

	// 	it("should expand the filter group when filters are selected despite it being set to collapse by default", () => {
	// 		const { container } = render(
	// 			<FilterGroup
	// 				{...filterGroupProps}
	// 				collapseByDefault={true}
	// 				selectedCount={1}
	// 			>
	// 				A filter group
	// 			</FilterGroup>
	// 		);
	// 		const fieldset = container.querySelector("fieldset");
	// 		expect(fieldset?.getAttribute("aria-hidden")).toBe("false");
	// 	});

	// 	it("should collapse the expanded filter group when the title is clicked", async () => {
	// 		const wrapper = render(
	// 			<FilterGroup {...filterGroupProps}>
	// 				<span>Filter group</span>
	// 			</FilterGroup>
	// 		);

	// 		const button = wrapper.getByRole("button");
	// 		userEvent.click(button);

	// 		await waitFor(() => {
	// 			const fieldset = wrapper.container.querySelector("fieldset");
	// 			expect(fieldset?.getAttribute("aria-hidden")).toBe("true");
	// 		});
	// 	});
	// });

	// describe("FilterOptions component", () => {
	// 	it("should pass the filter group heading down to the filter option and use it plus the child set the input id when no id or value is present", () => {
	// 		const wrapper = render(
	// 			<>
	// 				<FilterPanel {...filterPanelProps}>
	// 					<FilterGroup {...filterGroupProps}>
	// 						<FilterOption {...filterOptionProps}>First filter</FilterOption>
	// 					</FilterGroup>
	// 				</FilterPanel>
	// 			</>
	// 		);
	// 		const input = wrapper.getByRole("checkbox");
	// 		expect(input.id).toBe("filter_some-filters_first-filter");
	// 	});

	// 	it("should pass the filter group id down to the filter option and use it plus filter option value to set the input id when present ", () => {
	// 		const wrapper = render(
	// 			<>
	// 				<FilterPanel {...filterPanelProps}>
	// 					<FilterGroup {...filterGroupProps} id="group-id">
	// 						<FilterOption {...filterOptionProps} value="override-child">
	// 							First filter
	// 						</FilterOption>
	// 					</FilterGroup>
	// 				</FilterPanel>
	// 			</>
	// 		);
	// 		const input = wrapper.getByRole("checkbox");
	// 		expect(input.id).toBe("filter_group-id_override-child");
	// 	});

	// 	it("should fire an onChange function if one is supplied", async () => {
	// 		const wrapper = render(
	// 			<FilterOption {...filterOptionProps}>Second filter</FilterOption>
	// 		);
	// 		const checkbox = wrapper.getByRole("checkbox");
	// 		userEvent.click(checkbox);

	// 		await waitFor(() => {
	// 			expect(aFunction).toHaveBeenCalledTimes(1);
	// 		});
	// 	});
	// });

	describe("FilterPanel component", () => {
		it("should always have fallback properties present", async () => {
			const { container } = render(
				<FilterPanel {...filterPanelProps}>Second filter</FilterPanel>
			);
			const form = container.querySelector("form");
			// ...but why not use getByRole("form")?
			// This is why: https://github.com/testing-library/dom-testing-library/issues/474#issuecomment-597606894
			expect(form?.getAttribute("method")).toBe("GET");
			expect(form?.getAttribute("action")).toBe("/submit");
		});

		it("should cascade the heading level from the filter panel down to the filter group and filter by item components", () => {
			const { container } = render(
				<FilterPanel {...filterPanelProps} headingLevel={3}>
					<FilterGroup {...filterGroupProps}>
						<FilterOption {...filterOptionProps}>First filter</FilterOption>
						<FilterOption {...filterOptionProps}>Second filter</FilterOption>
					</FilterGroup>
					<FilterByInput label="something" name="something">
						Filter by name
					</FilterByInput>
				</FilterPanel>
			);

			const filterPanelHeading = container.querySelector(
				".filter-panel__heading"
			);
			expect(filterPanelHeading?.tagName).toEqual("H3");

			const filterGroupHeading = container.querySelector(
				".filter-group__heading"
			);
			expect(filterGroupHeading?.tagName).toEqual("H4");

			const filterByInputHeading = container.querySelector(
				".inputFilterBox__heading"
			);
			expect(filterByInputHeading?.tagName).toEqual("H4");
		});
	});
});
