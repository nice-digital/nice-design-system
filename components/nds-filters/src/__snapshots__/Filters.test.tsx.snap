// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@nice-digital/nds-filters Crash test should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filter-summary"
      >
        <div
          class="filter-summary__count"
        >
          <h2
            class="h5 mv--0"
          >
            Showing results 1 to 10 of 1209
          </h2>
        </div>
        <div
          class="filter-summary__sorting"
        >
          <label
            class="filter-summary__sort-label"
            for="sorting-options"
          >
            Sort by
          </label>
          <select
            class="filter-summary__select"
            id="sorting-options"
          >
            <option>
              Active
            </option>
            <option>
              Button
            </option>
          </select>
        </div>
        <ul
          aria-label="Active filters"
          class="filter-summary__filters hide-print"
        >
          <li
            class="filter-summary__filter"
          >
            <span
              class="tag tag--outline"
            >
              Another filter
              <button
                class="tag__remove"
              >
                <span
                  class="visually-hidden"
                >
                  Remove 
                  Another filter
                   filter
                </span>
              </button>
            </span>
          </li>
        </ul>
      </div>
      <form
        action="/submit"
        method="GET"
      >
        <div
          class="filter-panel"
        >
          <h2
            class="filter-panel__heading"
          >
            <button
              aria-controls="filter-panel-body"
              aria-expanded="true"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="filter-panel__heading-icon filter-panel__heading-icon--expanded"
                fill="none"
                focusable="false"
                height="1em"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26Z"
                  fill="currentColor"
                />
              </svg>
              Some filters
            </button>
          </h2>
          <div
            aria-hidden="false"
            class="filter-panel__body"
            id="filter-panel-body"
          >
            <div
              class="filter-group"
            >
              <h3
                class="filter-group__heading"
              >
                <button
                  aria-controls="group-some-filters"
                  aria-expanded="true"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="filter-group__heading-icon filter-group__heading-icon--expanded"
                    fill="none"
                    focusable="false"
                    height="1em"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26Z"
                      fill="currentColor"
                    />
                  </svg>
                  <span
                    id="group-heading-some-filters"
                  >
                    Some filters
                     
                  </span>
                </button>
              </h3>
              <fieldset
                aria-hidden="false"
                class="filter-group__options"
                id="group-some-filters"
              >
                <legend>
                  Some filters
                </legend>
                <label
                  class="filter-option"
                  for="filter_some-filters_a-filter"
                >
                  <input
                    checked=""
                    id="filter_some-filters_a-filter"
                    name="some-filters"
                    title="Some filters - A filter"
                    type="checkbox"
                    value="A filter"
                  />
                  <span
                    class="filter-option__text"
                  >
                    A filter
                  </span>
                </label>
              </fieldset>
            </div>
            <div
              class="inputFilterBox"
            >
              <h3
                class="inputFilterBox__heading"
              >
                <button
                  aria-controls="inputFilter-Filter by name"
                  aria-expanded="true"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="filter-group__heading-icon filter-group__heading-icon--expanded"
                    fill="none"
                    focusable="false"
                    height="1em"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26Z"
                      fill="currentColor"
                    />
                  </svg>
                  Filter by name
                </button>
              </h3>
              <div
                aria-hidden="false"
                class="inputFilterBox__controls"
                id="inputFilter-Filter by name"
              >
                <div
                  class="input"
                >
                  <label
                    class="input__label"
                    for="Filter by name"
                  >
                    Filter by name
                  </label>
                  <input
                    class="input__input"
                    id="Filter by name"
                    name="Filter by name"
                    type="text"
                    value=""
                  />
                </div>
                <button
                  class="btn ml--0 mb--0"
                  type="submit"
                >
                  Filter
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <div
      class="filter-summary"
    >
      <div
        class="filter-summary__count"
      >
        <h2
          class="h5 mv--0"
        >
          Showing results 1 to 10 of 1209
        </h2>
      </div>
      <div
        class="filter-summary__sorting"
      >
        <label
          class="filter-summary__sort-label"
          for="sorting-options"
        >
          Sort by
        </label>
        <select
          class="filter-summary__select"
          id="sorting-options"
        >
          <option>
            Active
          </option>
          <option>
            Button
          </option>
        </select>
      </div>
      <ul
        aria-label="Active filters"
        class="filter-summary__filters hide-print"
      >
        <li
          class="filter-summary__filter"
        >
          <span
            class="tag tag--outline"
          >
            Another filter
            <button
              class="tag__remove"
            >
              <span
                class="visually-hidden"
              >
                Remove 
                Another filter
                 filter
              </span>
            </button>
          </span>
        </li>
      </ul>
    </div>
    <form
      action="/submit"
      method="GET"
    >
      <div
        class="filter-panel"
      >
        <h2
          class="filter-panel__heading"
        >
          <button
            aria-controls="filter-panel-body"
            aria-expanded="true"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="filter-panel__heading-icon filter-panel__heading-icon--expanded"
              fill="none"
              focusable="false"
              height="1em"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26Z"
                fill="currentColor"
              />
            </svg>
            Some filters
          </button>
        </h2>
        <div
          aria-hidden="false"
          class="filter-panel__body"
          id="filter-panel-body"
        >
          <div
            class="filter-group"
          >
            <h3
              class="filter-group__heading"
            >
              <button
                aria-controls="group-some-filters"
                aria-expanded="true"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="filter-group__heading-icon filter-group__heading-icon--expanded"
                  fill="none"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26Z"
                    fill="currentColor"
                  />
                </svg>
                <span
                  id="group-heading-some-filters"
                >
                  Some filters
                   
                </span>
              </button>
            </h3>
            <fieldset
              aria-hidden="false"
              class="filter-group__options"
              id="group-some-filters"
            >
              <legend>
                Some filters
              </legend>
              <label
                class="filter-option"
                for="filter_some-filters_a-filter"
              >
                <input
                  checked=""
                  id="filter_some-filters_a-filter"
                  name="some-filters"
                  title="Some filters - A filter"
                  type="checkbox"
                  value="A filter"
                />
                <span
                  class="filter-option__text"
                >
                  A filter
                </span>
              </label>
            </fieldset>
          </div>
          <div
            class="inputFilterBox"
          >
            <h3
              class="inputFilterBox__heading"
            >
              <button
                aria-controls="inputFilter-Filter by name"
                aria-expanded="true"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="filter-group__heading-icon filter-group__heading-icon--expanded"
                  fill="none"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26Z"
                    fill="currentColor"
                  />
                </svg>
                Filter by name
              </button>
            </h3>
            <div
              aria-hidden="false"
              class="inputFilterBox__controls"
              id="inputFilter-Filter by name"
            >
              <div
                class="input"
              >
                <label
                  class="input__label"
                  for="Filter by name"
                >
                  Filter by name
                </label>
                <input
                  class="input__input"
                  id="Filter by name"
                  name="Filter by name"
                  type="text"
                  value=""
                />
              </div>
              <button
                class="btn ml--0 mb--0"
                type="submit"
              >
                Filter
              </button>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
